ast/edit/option_editor_test.go

```diff
@@ -0,0 +1,82 @@
+package edit_test
+
+import (
+	"testing"
+
+	"github.com/influxdata/flux/ast"
+	"github.com/influxdata/flux/ast/edit"
+	"github.com/influxdata/flux/parser"
+	"github.com/influxdata/flux/values"
+	"github.com/pkg/errors"
+)
+
+func TestEditor(t *testing.T) {
+	testCases := []struct {
+		name   string
+		in     string
+		edited string
+		editor *edit.OptionEditor
+	}{
+		{
+			name: "leaves_unchanged",
+			in:   `from(bucket:"testdb") |> range(start: 2018-05-23T13:09:22.885021542Z)`,
+			edited: `from(bucket: "testdb")
+	|> range(start: 2018-05-23T13:09:22.885021542Z)`,
+			editor: edit.NewOptionEditor(map[string]values.Value{
+				"bucket": values.New("foo"),       // should ignore this
+				"start":  values.New("yesterday"), // should ignore this
+				"stop":   values.New("tomorrow"),  // should ignore this
+			}),
+		},
+		{
+			name: "edits",
+			in: `option task = {name: "foo",every: 1h,delay: 10m,cron: "0 2 * * *",retry: 5}
+from(bucket:"testdb") |> range(start: 2018-05-23T13:09:22.885021542Z)`,
+			edited: `option task = {
+	name: "bar",
+	every: 7200000000000ns,
+	delay: 2520000000000ns,
+	cron: "buz",
+	retry: 10,
+}
+
+from(bucket: "testdb")
+	|> range(start: 2018-05-23T13:09:22.885021542Z)`,
+			editor: edit.NewOptionEditor(map[string]values.Value{
+				"bucket": values.New("foo"),       // should ignore this
+				"start":  values.New("yesterday"), // should ignore this
+				"name":   values.New("bar"),
+				"every":  values.New(duration("2h")),
+				"delay":  values.New(duration("42m")),
+				"cron":   values.New("buz"),
+				"retry":  values.New(int64(10)),
+			}),
```

Comment:
Not all options are objects, so these keys would not make sense in that case. For example the `now` option is actually a function. By default it is defined as:

```
option now = () => systemTime()
```


Or more simply the the `location` may need to be changed to a function call.

For example:

```
option location = loadLocation("Custom/Location")
```

Actually thinking about this I think that maybe the `values.Value` is the wrong interface to use, but rather we should use `ast.Expression`. Meaning you can replace an options `Init` with an arbitrary expression. Then we can make the `getLiteral` function public so that if a consumer has a values.Value they can create an ast.Expression from it.